src/Saturn/ControllerHelpers.fs

```diff
@@ -40,6 +42,46 @@ module ControllerHelpers =
     let file (ctx: HttpContext) path =
       ctx.WriteHtmlFileAsync path
 
+    ///Returns to the client response according to accepted content type (`Accept` header, and if it's not present `Content-Type` header)
+    let resposne (ctx: HttpContext) (output: 'a) =
+      let jsonMediaType = MediaTypeHeaderValue.Parse (StringSegment "application/json")
+      let xmlMediaType = MediaTypeHeaderValue.Parse (StringSegment "application/xml")
+      let plainMediaType = MediaTypeHeaderValue.Parse (StringSegment "text/plain")
+      let htmlMediaType = MediaTypeHeaderValue.Parse (StringSegment "text/html")
+
+      match ctx.Request.Headers.TryGetValue "Accept" with
+      | true, header ->
+        let mediaTypes = MediaTypeHeaderValue.ParseList header
+        if mediaTypes |> Seq.exists (jsonMediaType.IsSubsetOf) then ctx.WriteJsonAsync(output)
+        elif mediaTypes |> Seq.exists (xmlMediaType.IsSubsetOf) then ctx.WriteJsonAsync(output)
+        else
+          match typeof<'a> with
+          | k when k = typeof<string> && mediaTypes |> Seq.exists (plainMediaType.IsSubsetOf) -> ctx.WriteTextAsync(unbox<string> output)
```

Comment:
maybe we could expose a media-type-based plugging mechanism here? essentially a map of mediaType to `context -> output -> Task` (or whatever the expected output signature is)

I'm imagining people that use media types to influence serialization.